test/src/unit-cppapi-schema-evolution.cc

```diff
@@ -686,7 +686,211 @@ TEST_CASE(
+  bool sparse = array_type == TILEDB_SPARSE;
+
+  const char* layout_str = nullptr;
+  tiledb_layout_to_str(layout, &layout_str);
+  const char* type_str = nullptr;
+  tiledb_array_type_to_str(array_type, &type_str);
+  auto array_uri{
+      vfs_test_setup.array_uri("test_schema_evolution_query_condition_v2")};
+
+  DYNAMIC_SECTION(
+      type_str << " " << layout_str << " array"
+               << (duplicates ? " with duplicates enabled" : "")) {
+    {
+      ArraySchema schema(ctx, array_type);
+      // Duplicates are not supported for dense arrays.
+      if (sparse) {
+        schema.set_allows_dups(duplicates);
+        CHECK(duplicates == schema.allows_dups());
+      }
+      schema.set_cell_order(TILEDB_ROW_MAJOR);
+      schema.set_tile_order(TILEDB_COL_MAJOR);
+
+      Domain domain(ctx);
+      auto id1 = Dimension::create<int>(ctx, "d1", {{1, 4}}, 2);
+      auto id2 = Dimension::create<int>(ctx, "d2", {{1, 4}}, 2);
+      domain.add_dimension(id1).add_dimension(id2);
+      schema.set_domain(domain);
+
+      std::vector<std::string> enum_values = {"A", "B", "C", "D"};
+      auto e = Enumeration::create(ctx, "a_label", enum_values, false);
+      ArraySchemaExperimental::add_enumeration(ctx, schema, e);
+
+      auto a = Attribute::create<int>(ctx, "a");
+      AttributeExperimental::set_enumeration_name(ctx, a, "a_label");
+      schema.add_attribute(a);
+
+      auto b = Attribute::create<float>(ctx, "b");
+      float b_fill = -1.0f;
+      uint64_t size = sizeof(b_fill);
+      b.set_fill_value(&b_fill, size);
+      schema.add_attribute(b);
+
+      Array::create(array_uri, schema);
+    }
+
+    // Write data
+    {
+      Array array(ctx, array_uri, TILEDB_WRITE);
+      Query query(ctx, array, TILEDB_WRITE);
+      std::vector<int> d1_data = {1, 1, 2, 2};
+      std::vector<int> d2_data = {1, 2, 1, 2};
+      std::vector<int> a_data = {1, 2, 3, 4};
+      std::vector<float> b_data = {1.1, 2.2, 3.3, 4.4};
+
+      // Set coordinates.
+      if (sparse) {
+        query.set_data_buffer("d1", d1_data).set_data_buffer("d2", d2_data);
+      } else {
+        Subarray subarray(ctx, array);
+        subarray.add_range<int>(0, 1, 2).add_range<int>(1, 1, 2);
+        query.set_subarray(subarray);
+      }
+
+      // Set data buffers.
+      query.set_data_buffer("a", a_data).set_data_buffer("b", b_data);
+
+      // Perform the write and close the array.
+      CHECK(query.submit() == Query::Status::COMPLETE);
+      array.close();
+    }
+
+    // Evolve
+    {
+      // Drop attribute 'a'.
+      uint64_t now = tiledb_timestamp_now_ms() + 1;
+      ArraySchemaEvolution schemaEvolution = ArraySchemaEvolution(ctx);
+      schemaEvolution.set_timestamp_range(std::make_pair(now, now));
+      schemaEvolution.drop_attribute("a").array_evolve(array_uri);
+
+      // Add attribute 'a' without an enumeration label.
+      // Also modify it's datatype from fixed int to var-size string.
```

Comment:
Can you add this kind of commentary on what the test is doing as a method comment?

Reply:
Done